### PR TITLE
add old and new balances

### PR DESCRIPTION
**Pull request by fatemebagherii**

### CODE REVIEW

File: cmd/rpcdaemon/commands/eth_get_balance_changes.go [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+		fmt.Println("old balance: ", balances.oldBalance)
+		fmt.Println("new balance: ", balances.newBalance)
+	}
+
+	return nil
+}
+
+func (api *APIImpl) GetAccountNewBalance(address common.Address, blockNrOrHash rpc.BlockNumberOrHash) uint256.Int {
+
+	ctx, _ := common2.RootContext()
```
Comment (AskAlexSharov) [inline cmd/rpcdaemon/commands/eth_get_balance_changes.go:99]: Use context of request if you don’t want run something in background

File: cmd/rpcdaemon/commands/eth_get_balance_changes.go [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+	}
+	return true
+}
+
+type oldNewBalance struct {
+	oldBalance *hexutil.Big
+	newBalance *hexutil.Big
+}
+
+func (api *APIImpl) GetChangeSet(blockNum uint64) map[common.Address]oldNewBalance {
```
Comment (AskAlexSharov) [inline cmd/rpcdaemon/commands/eth_get_balance_changes.go:36]: Geth has this api method?

File: cmd/rpcdaemon/commands/eth_get_balance_changes.go [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+	for address, balances := range mapping {
+		fmt.Println("address: ", address)
+		fmt.Println("old balance: ", balances.oldBalance)
+		fmt.Println("new balance: ", balances.newBalance)
+	}
+
+	return nil
+}
+
+func (api *APIImpl) GetAccountNewBalance(address common.Address, blockNrOrHash rpc.BlockNumberOrHash) uint256.Int {
```
Comment (AskAlexSharov) [inline cmd/rpcdaemon/commands/eth_get_balance_changes.go:97]: Geth has this method? If no - let’s move it to erigon_ namespace

File: cmd/rpcdaemon/commands/eth_get_balance_changes.go [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+
+	c, err := tx.Cursor("AccountChangeSet")
+	if err != nil {
+		fmt.Println(err)
+	}
+	defer c.Close()
+	startkey := dbutils.EncodeBlockNumber(blockNum)
+
+	decode := changeset.Mapper["AccountChangeSet"].Decode
+	fmt.Println("block number: ", startkey)
```
Comment (AskAlexSharov) [inline cmd/rpcdaemon/commands/eth_get_balance_changes.go:53]: Debug line

**Review by AskAlexSharov — CHANGES_REQUESTED** [review]

File: cmd/rpcdaemon/commands/eth_get_balance_changes.go [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+
+	balancesMapping := make(map[common.Address]oldNewBalance)
+
+	for dbKey, dbValue, _ := c.Seek(startkey); reflect.DeepEqual(dbKey, startkey) && dbKey != nil; dbKey, dbValue, _ = c.Next() {
```
Comment (AlexeyAkhunov) [inline cmd/rpcdaemon/commands/eth_get_balance_changes.go:57]: Not sure why `reflect.DeepEqual` is needed here to compare slice of bytes?

File: cmd/rpcdaemon/commands/eth_get_balance_changes.go [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+
+		_, address, v, err := decode(dbKey, dbValue)
+		if err != nil {
+			fmt.Println(err)
```
Comment (AlexeyAkhunov) [inline cmd/rpcdaemon/commands/eth_get_balance_changes.go:61]: Not a good way to deal with errors here

File: cmd/rpcdaemon/commands/eth_get_balance_changes.go [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+		}
+		old_balance := (*hexutil.Big)(acc.Balance.ToBig())
+		var commonAddress common.Address
+		copy(commonAddress[:], address[:20])
+		n := rpc.BlockNumber(blockNum)
+		new_balance_slice := api.GetAccountNewBalance(commonAddress, rpc.BlockNumberOrHash{BlockNumber: &n})
+		new_balance := (*hexutil.Big)(new_balance_slice.ToBig())
+
+		if !reflect.DeepEqual(old_balance, new_balance) {
```
Comment (AlexeyAkhunov) [inline cmd/rpcdaemon/commands/eth_get_balance_changes.go:74]: There is a better way to compare the values of this type

File: cmd/rpcdaemon/commands/eth_get_balance_changes.go [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+	"github.com/ledgerwatch/erigon/common"
+	"github.com/ledgerwatch/erigon/common/changeset"
+	"github.com/ledgerwatch/erigon/common/dbutils"
+	"github.com/ledgerwatch/erigon/common/hexutil"
+	"github.com/ledgerwatch/erigon/core/types/accounts"
+	"github.com/ledgerwatch/erigon/rpc"
+	"github.com/ledgerwatch/erigon/turbo/rpchelper"
+)
+
+func bytesEqual(a, b []byte) bool {
```
Comment (AlexeyAkhunov) [inline cmd/rpcdaemon/commands/eth_get_balance_changes.go:19]: What about `bytes.Equal` ?

**Pull request closed without merging**